### DISCUSSION TITLE
m2a: is there a documentation or tutorial?

### DISCUSSION THREAD

**Discussion by thefrontendguy** [directus/directus#11068] [Archive]
I tried out the m2a relations. Everything worked (I could get the data from the api). As long I did not reorder the items. Then I would get null or {} for the reordered item.

1. is there a way to show the preview (instead of opening the drawer)?
2. when I try to reorder, i get an error message. probably because the setup was wrong?

It would be nice to have a step by step tutorial with best practices, how to setup up m2a. Also with translations.

See the attachments.

[pages](https://user-images.githubusercontent.com/32142588/149627528-5968cc9a-9b82-4ce3-81ed-9cc05b765eae.png)

[error after reordering the item](https://user-images.githubusercontent.com/32142588/149627531-ac893896-9b60-44bf-9d73-693feb09b59b.png)

**Comment by joselcvarela** [comment]
Hello @thefrontendguy 
What database are you using?

**Reply by thefrontendguy** [reply]
PostgreSQL 13.5 (Docker)
directus/directus:9.4 (Docker)

**Reply by joselcvarela** [reply]
From what I see on your screenshot you also have a drawer opened.
Can you share a short clip demonstrating the issue? I checked M2A reorder and everything is working as expected, so I think the issue may come from the place you are trying to do that operation

**Reply by thefrontendguy** [reply]
You can watch it here:

[video](https://kayacahit-directus-cms.apps.my-paas.de/assets/3ba551e1-c091-4786-8f2c-117d1c16a029.mp4)

The last part of the video is missing.
This is the graphql output:
`
{
   "data": {
    "pages_by_id": {
      "id": "10",
      "title": "the page title",
      "status": "draft",
      "m2a": [
        {
          "item": null
        },
        {
          "item": null
        },
        {
          "item": null
        }
      ]
    }
  }
}`

**Reply by joselcvarela** [reply]
Huumm the issue seems to be inside the `Block P: 5` and not related with  the M2A.
Is it ok to share your schema (not data)? If so, can you run this command `npx directus schema snapshot ./snapshot.yaml ` and attach the `snapshot.yaml` here? Maybe this is a misconfiguration of the table 🤔

**Reply by thefrontendguy** [reply]
[snapshot.yaml](https://kayacahit-directus-cms.apps.my-paas.de/assets/158450d5-21a1-433a-abd8-3ad213b91f5a?download)

**Reply by thefrontendguy** [reply]
I testet similar things with m2m. This works as expected. I see the preview text and also the reordering works fine.

**Comment by reqwire** [comment]
can we please get some documentation or tutorials on how to set up relationships properly in directus :(

I cant make bidirectional M2M or M2A no matter how hard I try. Always one of them says it's misconfigured

**Reply by reqwire** [reply]
EDIT: After pulling my hair out I figured out you have to go into advanced new field mode to set it up bidirectionally.

**Reply by rijkvanzanten** [reply]
The docs are open source as well, so all feedback you have / edits on those are much appreciated at https://github.com/directus/docs 🙂
